File: backend/JobHunter.Domain/Entities/Company.cs
using System.Text.Json;
using JobHunter.Domain;

namespace JobHunter.Domain.Entities;

public partial class Company : BaseEntity
{
    public string? Name { get; set; }

    public string? WebsiteUrl { get; set; }

    public string? Country { get; set; }

    public string? CompanyType { get; set; }

    public string? LogoUrl { get; set; }

    public string? CoverPhotoUrl { get; set; }

    public string? Overview { get; set; }

    public string? Benefits { get; set; }

    public JsonDocument? TeamPhotoUrls { get; set; }

    public string? TeamSize { get; set; }

    public string? Slug { get; set; }

    public virtual ICollection<CompanyBranch> CompanyBranches { get; set; } = new List<CompanyBranch>();

    public virtual ICollection<Job> Jobs { get; set; } = new List<Job>();

    public virtual ICollection<User> Users { get; set; } = new List<User>();
}

File: frontend/src/types/company.ts
import type { CompanyBranchResponse } from "./job"

export interface CompanyCard {
  id: string
  name: string
  slug: string
  logoUrl: string | null
  coverPhotoUrl: string | null
  companyType: string | null
  teamSize: string | null
  country: string | null
  openingVacancies: number
  numberOfFollowers: number
}

export interface Company {
  id: string
  name: string
  websiteUrl: string | null
  country: string | null
  companyType: string | null
  logoUrl: string | null
  coverPhotoUrl: string | null
  overview: string | null
  benefits: string | null
  teamPhotoUrls: string[]
  teamSize: string | null
  slug: string
  companyBranches: CompanyBranchResponse[]
}
import { File } from "buffer";

export interface TeamImages {
  urls: File[]
}

export interface BrandingResponseDto extends BrandingRequestDto {

    teamPhotoUrls?: string[];
}

export interface BrandingRequestDto {
  overview?: string;
  benefits?: string;
}

export interface CompanyGeneralRequestDto {
  name?: string;
  teamSize?: string;
  websiteUrl?: string;
  country?: string;
  companyType?: string;
}

export interface CompanyGeneralResponseDto extends CompanyGeneralRequestDto {
  logoUrl?: string;
  coverUrl?: string;
}


export interface CompanyBranchReponseDto {
  name: string;
  address:string;
  city: string;
}

export interface CompanyBranchRequestDto extends CompanyBranchReponseDto {
  id: string;
}